### PR TITLE
Dark mode compatibility with gnome

### PR DESCRIPTION
**Pull request by arnaudTriolet**

### CODE REVIEW

**Comment by arnaudTriolet** [comment]
Hi,
I was using that lib in one of my project, and i recently switch to Gnome Desktop and I realize that it wasn't dealing with gnome dark theme because of a different gconf path.
So I made that patch.
Hope it will be integrated.
Hope you'll like it

**Comment by sindresorhus** [comment]
Please manually ensure that this code addition works in practice.

File: source/linux/background-managers/gnome.js [hunk 1]
```diff
@@ -4,21 +4,42 @@ export async function isAvailable() {
...
-		'set',
-		'org.gnome.desktop.background',
-		'picture-uri',
-		`file://${imagePath}`,
-	]);
+	if (isDarkTheme()) {
+		await execFile('gsettings', [
+			'set',
+			'org.gnome.desktop.background',
+			'picture-uri-dark',
```
Comment (sindresorhus) [inline source/linux/background-managers/gnome.js:34]: The only difference is this, so it can just be a ternary here instead of an `if`.

Comment (arnaudTriolet) [reply source/linux/background-managers/gnome.js]: I changed it on my last commit

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: source/linux/background-managers/gnome.js [hunk 2]
```diff
@@ -4,11 +4,23 @@ export async function isAvailable() {
...
+		'get',
+		'org.gnome.desktop.interface',
+		'color-scheme',
+	]);
+
+	return Boolean(themeStyle === 'prefer-dark');
+}
+
 export async function get() {
+	const prop = (isDarkTheme()) ? 'picture-uri-dark' : 'picture-uri';
```
[the remark this answers is not in the document]
Comment (arnaudTriolet) [reply source/linux/background-managers/gnome.js:18]: WTF you are right, but the test was doing the job... I don't understand why but... OK

**Comment by arnaudTriolet** [comment]
I think we should be ok with the last commit

File: source/linux/background-managers/gnome.js [hunk 3]
```diff
@@ -18,7 +30,7 @@ export async function set(imagePath) {
 	await execFile('gsettings', [
 		'set',
 		'org.gnome.desktop.background',
-		'picture-uri',
+		(await isDarkTheme()) ? 'picture-uri-dark' : 'picture-uri',
```
Comment (sindresorhus) [inline source/linux/background-managers/gnome.js:33]: Move it to a variable, for consistency with `get`.

Comment (arnaudTriolet) [reply source/linux/background-managers/gnome.js:33]: Do it yourself

**Pull request closed without merging**